aws_lambda_powertools/utilities/idempotency/base.py

```diff
@@ -0,0 +1,181 @@
+import logging
+from typing import Any, Callable, Dict, Optional, Tuple
+
+from aws_lambda_powertools.utilities.idempotency.config import IdempotencyConfig
+from aws_lambda_powertools.utilities.idempotency.exceptions import (
+    IdempotencyAlreadyInProgressError,
+    IdempotencyInconsistentStateError,
+    IdempotencyItemAlreadyExistsError,
+    IdempotencyItemNotFoundError,
+    IdempotencyKeyError,
+    IdempotencyPersistenceLayerError,
+    IdempotencyValidationError,
+)
+from aws_lambda_powertools.utilities.idempotency.persistence.base import (
+    STATUS_CONSTANTS,
+    BasePersistenceLayer,
+    DataRecord,
+)
+
+MAX_RETRIES = 2
+logger = logging.getLogger(__name__)
+
+
+class IdempotencyHandler:
+    """
+    Base class to orchestrate calls to persistence layer.
+    """
+
+    def __init__(
+        self,
+        function: Callable,
+        function_payload: Any,
+        idempotency_config: IdempotencyConfig,
```

Comment:
@heitorlessa  sometimes this is `config` (existing code) and sometimes it is `idempotentcy_config`. Should we consistent? Or is this a matter of a better style?

Reply:
Good catch. I think this is a left over from the many refactoring attempts I had at it, lemme correct it to `config`